### PR TITLE
 Add assertText assertion to check that a given text is shown in a given View

### PR DESCRIPTION
**Pull request by EsteveAguilera**
Add assertion 
`assertText(R.id.widget, "expected text")`

This assertion was requested on issue #120

### CODE REVIEW

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java [hunk 1]
```diff
@@ -305,4 +311,9 @@ public void checkViewHasNotFocus() throws Exception {
     assertNotFocused(R.string.edittext_with_no_focus);
     assertNotFocused("EditText with no focus");
   }
+
+  @Test
+  public void checkExpectedText() throws Exception {
```
Comment (rocboronat) [inline sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: A small comment there: can you move this test near the other test methods that check for texts? Just to have them grouped 🙌

**Comment by Sloy** [comment]
Hmm I wonder: if we use [assertDisplayed(text)`, why changing the naming instead of using `assertDisplayed(id, text)`? Do we need to make a distinction?

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java (hunk 1 above)
Comment (EsteveAguilera) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: Of course!

**Comment by rocboronat** [comment]
+1 to @Sloy. `assertDisplayed` sounds like the method that checks for everything that it displayed, and we could offer lots of it.

**Comment by EsteveAguilera** [comment]
Ok, two owners is a KO win!

**Comment by rocboronat** [comment]
I proposed the bad name at #120, so... one of the owners is the guilty 🍶

**Comment by EsteveAguilera** [comment]
Ooops, I just realized I've been working directly on my **barista2** branch instead of a new one for the PR... sorry about that! 😳 

The last commit also moves the test @rocboronat, I hope it's on a better place now 😅

**Comment by rocboronat** [comment]
That's the Fewlaps performance! 💪

**Comment by rocboronat** [comment]
That's the Android Studio autoformatter performance! 💪

File: library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt [hunk 2]
```diff
@@ -40,6 +40,11 @@ object BaristaAssertions {
         assertDisplayed(withText(text))
     }
 
+    @JvmStatic
+    fun assertDisplayed(@IdRes resId: Int, text: String) {
+        onView(resId.resourceMatcher()).check(matches(withText(text)));
```
Comment (Sloy) [inline library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt]: Watch out! Here you don't want to use our magic (I think). 
`resourceMatcher()` is intended to give you the matcher for an id **or** a string resource.

That means you can use `assertDisplayed(R.string.hello, "hello")`. I think it's best that we limit the matcher to `withId(id)`, so users won't get unexpected results if they use the API wrong.

Comment (rocboronat) [reply library/src/main/java/com/schibsted/spain/barista/assertion/BaristaAssertions.kt]: Uuuuuuh it's the **EDD** time!

File: sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java [hunk 3]
```diff
@@ -64,6 +64,11 @@ public void checkVisible_withRepeatedViews() throws Exception {
   }
 
   @Test
+  public void checkExpectedText() throws Exception {
```
[the remark this answers is not in the document]
Comment (EsteveAguilera) [reply sample/src/androidTest/java/com/schibsted/spain/barista/sample/AssertionsTest.java]: 👍

**Comment by EsteveAguilera** [comment]
😊

**Pull request merged**